twake/backend/node/src/services/channels/provider.ts

```diff
@@ -2,4 +2,12 @@ import { CRUDService } from "../../core/platform/framework/api/crud-service";
 import { TwakeServiceProvider } from "../../core/platform/framework/api";
 import { Channel } from "./entities";
 
-export default interface ChannelServiceAPI extends TwakeServiceProvider, CRUDService<Channel> {}
+export type ChannelPrimaryKey = {
+  id?: string;
+  company_id?: string;
+  workspace_id?: string;
+};
```

Comment:
How do you define the primary key order ?
In Cassandra the primary key contains a partition key and clustering keys. For instance in this case we have:
Partition key = [company_id]
Clustering key = [workspace_id, id]
In cassandra this is defined like this: ((company_id), workspace_id, id).

>It means that we cannot access the resources using:
company_id and id,
workspace_id and id,
id alone
workspace_id alone
>We only can access resources using:
company_id, company_id and workspace_id, company_id and workspace_id and id

The clustering keys also contain sorting information.

So we need to be able to define all this information :)

Reply:
Just tell me how to do this. I guess this is defined when we create the table, and then when we build the queries.

Reply:
Yes, I'm used to have the table generation done by the ORM.
Could we have this information here anyway even if node will not generate the tables ?
Something like:
```
@Entity(() => {
    table: "channel",
    primaryKey: [["company_id"], "workspace_id", "id"]
})
export class Channel {
```

Reply:
Yes, we will but not in this PR, it starts to be huge. I will check how we can automate this, if we can. If not we will provide scripts.